### PR TITLE
Update main installation method to OperatorHub install, testsuite cleanup 

### PR DESCRIPTION
**Pull request by AnetaCadova**

### CODE REVIEW

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoOperatorhub.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+@Slf4j
+
+public class ApicuritoOperatorhub extends ApicuritoInstall {
+    public static final String SUBSCRIPTION_NAME = "fuse-apicurito";
+
+    public static void deploy(String updateChannel, String crName) {
+        log.info("Deploying using Operatorhub");
+
+        Bundle.createSubscription(ConfigurationOCPSteps.getOpenShiftService(), SUBSCRIPTION_NAME,
```
Comment (mmajerni) [inline src/test/java/apicurito/tests/configuration/templates/ApicuritoOperatorhub.java:26]: So it is not possible or very time demanding to get rid of marketplace-utilities?

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoInstall.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ public static void cleanNamespace() {
         TestConfiguration.printDivider("Deleting namespace resources");
 
         try {
-            OpenShiftUtils.getInstance().customResourceDefinitions().withName("apicuritoes.apicur.io").delete();
```
Comment (mmajerni) [inline src/test/java/apicurito/tests/configuration/templates/ApicuritoInstall.java:26]: why this change? The old method is not working well?

**Review by mmajerni — CHANGES_REQUESTED** [review]
Also, please update README based on changes in test.properties and also changes in installations.

File: src/test/resources/catalogsource.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# run `oc apply -f catalogsource.yaml -n openshift-marketplace` to apply this catalogsource
```
[the remark this answers is not in the document]
Comment (AnetaCadova) [reply src/test/resources/catalogsource.yaml:1]: It should be handled by OCP, this is a backup, so we can do it manually too.

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoInstall.java (hunk 2 above)
Comment (AnetaCadova) [reply src/test/java/apicurito/tests/configuration/templates/ApicuritoInstall.java:26]: The old method didn't delete the CRD

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoOperatorhub.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            updateChannel, "''",
+            TestConfiguration.getApicuritoIibImage());
```
[the remark this answers is not in the document]
Comment (AnetaCadova) [reply src/test/java/apicurito/tests/configuration/templates/ApicuritoOperatorhub.java:27]: Yes, it works with IIB images as we planned.

File: src/test/java/apicurito/tests/utils/slenide/ConfigurationOCPUtils.java [hunk 5]
```diff
@@ -3,48 +3,72 @@
...
-                .getStatus().getUnavailableReplicas();
+    public static void waitForOperatorUpdate() {
+        try {
+            OpenShift ocp = ConfigurationOCPSteps.getOpenShiftService().getClient();
+            String previousCSV = ReleaseSpecificParameters.APICURITO_OLD_CSV;
+            String newCSV = ReleaseSpecificParameters.APICURITO_CURRENT_CSV;
+            Filter completeFilter = Filter.filter(Criteria.where("phase").is("Complete"));
+            Filter matchesCSVs = Filter.filter(Criteria.where("bundleLookups.identifier").eq(newCSV).and("bundleLookups.replaces").eq(previousCSV));
+            HelperFunctions.waitFor(() -> {
+                DocumentContext documentContext = JsonPath.parse(ocp.customResource(installPlanContext()).list(ocp.getNamespace()));
```
[the remark this answers is not in the document]
Comment (AnetaCadova) [reply src/test/java/apicurito/tests/utils/slenide/ConfigurationOCPUtils.java:74]: We used exactly the same method before, so hopefully not

File: src/test/resources/catalogsource.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mmajerni) [reply src/test/resources/catalogsource.yaml:1]: OK, we can have it in code but please change private unreleased image to some text. Something like:
  image: registry-proxy.engineering.redhat.com/rh-osbs/iib:263984 --> image: "CHANGE ME TO IIB IMAGE"

File: src/test/java/apicurito/tests/steps/ConfigurationOCPSteps.java [hunk 6]
```diff
@@ -263,52 +192,11 @@ public void checkThatMeteringLabelsHaveCorrectValues(Component component) {
         }
     }
 
-    @Then("check that name and image of operator in operatorhub are correct")
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply src/test/java/apicurito/tests/steps/ConfigurationOCPSteps.java]: Since we are deleting operatohub test. Could we create another test or move this test to some existing?

File: src/test/java/apicurito/tests/steps/ConfigurationOCPSteps.java [hunk 7]
```diff
@@ -146,57 +153,36 @@ public void checkThatApicuritoOperatorIsDeployedAndInRunningState() {
...
+                try {
+                    ocp.customResource(ApicuritoOperatorhub.getSubscriptionContext()).edit(namespace, SUBSCRIPTION_NAME, subscription.toString());
+                    return true;
+                } catch (Exception e) {
+                    return false;
+                }
+            }, 300L, 120L);
+            ConfigurationOCPUtils.waitForOperatorUpdate();
+            ConfigurationOCPUtils.waitForPodsUpdate(2);
+        } catch (Exception e) {
```
Comment (mmajerni) [inline src/test/java/apicurito/tests/steps/ConfigurationOCPSteps.java:182]: waitFor method throws these two exceptions: `InterruptedException` `TimeoutException`. Please catch them instead of `Exception`

File: README.md [hunk 8]
```diff
@@ -82,6 +80,11 @@ apicurito.config.quay.pull.secret=
   - quay.io/redhat/apicurito-operator:1.8-x
   - docker.io/redhat/apicurito-operator:1.8-x
 
+**apicurito.config.commit.hash** specifies the version of the https://github.com/jboss-fuse/apicurio-operators repository.
+- Needed only for Operator installation.
+- Default is master.
+- You can find the commit hash in the operator image build log.
+ (e.g. http://download.eng.bos.redhat.com/brewroot/packages/fuse-apicurito-rhel-8-operator-container/1.11/21/data/logs/orchestrator.log)
```
Comment (mmajerni) [inline README.md:87]: An empty line is needed after this one to separate the two parameters

**Review by mmajerni — APPROVED** [review]
LGTM
